lib/bolt/fiber_executor.rb

```diff
@@ -101,12 +105,54 @@ def round_robin
     # Bolt can exit.
     #
     def plan_complete?
-      plan_futures.empty?
+      active_futures.empty?
+    end
+
+    def all_futures
+      active_futures + finished_futures
+    end
+
+    # Get the PlanFuture object that is currently executing
+    #
+    def get_current_future(fiber:)
+      # If no future is found, then this can just be nil.
```

Comment:
Hm, I can delete the comment. I was worried about matching with multiple Futures or no Futures (so much that initially I had a check for those cases as part of this function). That's true right now in Bolt, but if this gets used in other contexts (ie. PE) it might not always be true. I think matching multiple Futures is actually a much bigger issue if it ever does happen than matching no Futures, hence the comment, but....we can deal with that if it ever actually becomes an issue which seems unlikely.